src/Application/Adapters/WordPress/Preset.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Application\Adapters\WordPress;
+
+use NunoMaduro\PhpInsights\Domain\Contracts\Preset as PresetContract;
+use NunoMaduro\PhpInsights\Domain\Insights\ForbiddenDefineFunctions;
+use NunoMaduro\PhpInsights\Domain\Insights\ForbiddenUsingGlobals;
+use PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff;
+use SlevomatCodingStandard\Sniffs\Namespaces\AlphabeticallySortedUsesSniff;
+use SlevomatCodingStandard\Sniffs\TypeHints\DeclareStrictTypesSniff;
+use SlevomatCodingStandard\Sniffs\TypeHints\TypeHintDeclarationSniff;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * @internal
+ */
+final class Preset implements PresetContract
+{
+    /**
+     * {@inheritDoc}
+     */
+    public static function getName(): string
+    {
+        return 'wordpress';
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function get(): array
+    {
+        return [
+            'exclude' => [
+                'wp-admin',
+                'wp-includes',
+                'index.php',
+            ],
+            'remove'  => [
+                AlphabeticallySortedUsesSniff::class,
+                DeclareStrictTypesSniff::class,
+                ForbiddenDefineFunctions::class,
+                TypeHintDeclarationSniff::class,
+                ForbiddenUsingGlobals::class
+            ],
+            'config'  => [
+                ForbiddenFunctionsSniff::class => [
+                    'forbiddenFunctions' => [
+                        'var_dump' => null,
+                    ],
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function shouldBeApplied(array $composer): bool
+    {
+        /** @var array<string, string> $requirements */
+        $requirements = $composer['require'] ?? [];
+
+        if (count($requirements) > 0) {
+            return self::composerDiscovery($requirements);
+        }
+
+        return self::manualInstallationDiscovery();
+    }
+
+    /**
+     * Defining ways to discover WordPress through composer.
+     *
+     * @param  array<string, string>  $requirements  Composer requirements list.
+     *
+     * @return bool
+     */
+    private static function composerDiscovery(array $requirements): bool
+    {
+        foreach (array_keys($requirements) as $requirement) {
+            $requirement = (string) $requirement;
+
+            if (strpos($requirement, 'johnpbloch/wordpress') !== false) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * When manual installation is been use.
+     *
+     * @return bool
+     */
+    private static function manualInstallationDiscovery(): bool
+    {
+        $finder = new Finder();
+
+        $finder
+            ->in(__DIR__)
```

Comment:
Wrong this, you should use the $dir provided by the application.

Reply:
Well, is kinda tricky on WordPress not such of .env or so just the constant `ABSPATH` that might be the one you are making reference.

Reply:
The current `$dir` it's on the AnalyseCommand. You just need to make sure you can get it in your wordpress preset. Basically, you should pass that argument.

Reply:
🤔 I'm looking the best way of doing this. The only way is to pass the $directory to the `$presetClass::shouldBeApplied($directory, $composer);` and from there, I could pass that directory to the manualDiscovery ... 

Also I detect that sometimes, on AnalyseCommand.php the directory is a string still tho be treated as an Object Class

`$results = $this->analyser->analyse($style, $this->getConfig($input, $directory), (string) $directory);` This is what I'm doing, any idea / thought why of this ?

Reply:
could be this mark as a Done or shall we refactor the way of passing the directory ?

Reply:
We should pass the current directory yes.

Reply:
I thought the latest code does that